test/integration/network_extension_discovery_integration_test.cc

```diff
@@ -0,0 +1,801 @@
+
+  // Send data without send config update.
+  sendDataVerifyResults(default_bytes_to_drain_);
+
+  // Send update should cause a different response.
+  sendXdsResponse(filter_name_, "1", 3);
+  test_server_->waitForCounterGe(
+      "extension_config_discovery.network_filter." + filter_name_ + ".config_reload", 1);
+  sendDataVerifyResults(3);
+}
+
+TEST_P(NetworkExtensionDiscoveryIntegrationTest, BasicWithoutWarmingConfigFail) {
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  addFilterChain();
+  addDynamicFilter(filter_name_, true);
+  initialize();
+
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 1);
+
+  // Send data without send config update.
+  sendDataVerifyResults(default_bytes_to_drain_);
+
+  // Send config update with invalid config (drain_bytes has to >=2).
+  sendXdsResponse(filter_name_, "1", 1);
+  test_server_->waitForCounterGe(
+      "extension_config_discovery.network_filter." + filter_name_ + ".config_fail", 1);
+  sendDataVerifyResults(default_bytes_to_drain_);
+}
+
+TEST_P(NetworkExtensionDiscoveryIntegrationTest, TwoSubscriptionsSameName) {
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  addFilterChain();
+  addDynamicFilter(filter_name_, true);
+  addDynamicFilter(filter_name_, false);
+  initialize();
+
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 1);
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initializing);
+
+  sendXdsResponse(filter_name_, "1", 3);
+  test_server_->waitForCounterGe(
+      "extension_config_discovery.network_filter." + filter_name_ + ".config_reload", 1);
+
+  // Each filter drain 3 bytes.
+  sendDataVerifyResults(6);
+}
+
+TEST_P(NetworkExtensionDiscoveryIntegrationTest, TwoSubscriptionsDifferentName) {
+  two_connections_ = true;
+  on_server_init_function_ = [&]() { waitXdsStream(); };
+  addFilterChain();
+  addDynamicFilter("foo", true);
+  addDynamicFilter("bar", false, true, false, true);
+  initialize();
+
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 1);
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initializing);
+
+  // Send 1st config update.
+  sendXdsResponse("foo", "1", 3);
+  sendXdsResponse("bar", "1", 4, false, true);
+  test_server_->waitForCounterGe("extension_config_discovery.network_filter.foo.config_reload", 1);
+  test_server_->waitForCounterGe("extension_config_discovery.network_filter.bar.config_reload", 1);
+  // The two filters drain 3 + 4  bytes.
+  sendDataVerifyResults(7);
+
+  // Send 2nd config update.
+  sendXdsResponse("foo", "2", 4);
+  sendXdsResponse("bar", "2", 5, false, true);
+  test_server_->waitForCounterGe("extension_config_discovery.network_filter.foo.config_reload", 2);
+  test_server_->waitForCounterGe("extension_config_discovery.network_filter.bar.config_reload", 2);
+  // The two filters drain 4 + 5  bytes.
+  sendDataVerifyResults(9);
+}
+
```

Comment:
I think it is valuable to add integration test cases for pure static filters as this:

https://github.com/envoyproxy/envoy/blob/06e3e7d59fddc6541cecf399a8adc64092df7ed8/test/integration/listener_extension_discovery_integration_test.cc#L496

Reply:
@yanjunxiang-google can you please elaborate on the value that you see? Is it not just like not using ECDS at all (which I believe is covered in regular network filter tests)?